website/blog/2024-04-11-thalium.md

```diff
@@ -0,0 +1,619 @@
+types within a union type.
+
+![union](https://i.imgur.com/yY2TvSk.gif)
+
+## Debugger improvements
+
+Scala debugger adapter library, which is responsible for providing debugging
+experience in Metals, has been released recently under v4.0.0 with a lot of
+great improvements. These include:
+
+- Pretty stacktraces (Scala 3 Only) - improved stack trace names that are easier
+  and more useful to read for the users.
+
+![pretty](https://github.com/scalacenter/scala-debug-adapter/assets/13123162/8e0cbfe0-26ea-418e-913c-43400be04d37)
+
+- Runtime evaluator - allows evaluating Java-like expression in Scala and Java
+  source files without using the compiler. It is faster, it can access private
+  members, and it does not require casting to access members of runtime types.
+- Hot-code replace - The hot-code replace feature allows to change the behavior
+  of a running program. It is now possible to change the body of a method, and
+  to reload it in the debugged program. After the hot-code replacement, any call
+  to the modified method will execute the new implementation, even on former
+  instances of the changed class. This feature is particularly useful to track a
+  bug and fix it in a single debug session.
+
+![hot-code](https://github.com/scalacenter/scala-debug-adapter/assets/13123162/f5dbce3a-ca64-484d-a377-e52c091e3f05)
+
+- Collapse skipped frames - the debugger indicated frames which are not useful
+  for the users, Metals will now collapse them by default.
+
+![collapse](https://github.com/scalameta/metals/assets/13123162/2923f1e9-3ec6-4240-891b-36593434dc48)
+
+This all was done thanks to the hard work of [adpi2](https://github.com/adpi2),
+[iusildra](https://github.com/iusildra) and other contributors.
+
+You can read more on the
+[v4.0.0 release page](https://github.com/scalacenter/scala-debug-adapter/releases/tag/v4.0.0)
+
+## Go to implementations for dependency sources
+
+Previously, it was only possible to search for implementations inside
+dependencies as Metals didn't index type hierarchy. Thanks to
+[kasiaMarek](https://github.com/kasiaMarek) it's now possible to do that in both
+Scala and Java files with an exception of JDK to avoid additional cost of
+indexing it in every workspace. The latter will be fixed later on.
+
+![implementations](https://i.imgur.com/JarfZHE.gif)
+
+As with inlay hints, this feature will not work when using older deprecated
+Scala 2 versions and some of the older Scala 3 3.3.x and 3.4.x versions.
+
+## Completions for implicit classes
+
+Another feature implemented by [kasiaMarek](https://github.com/kasiaMarek) is
+adding support for completions of old style extension methods that were declared
+within implicit classes. This will work similar to extensions method completions
+for Scala 3.
+
+![implicit](https://i.imgur.com/pLt8GF2.gif)
+
+## Plugin author changes
+
+Doctor format version changed to 6 due to the added project Java information.
```

Comment:
Maybe add link to PR so it's easy to find what changed: https://github.com/scalameta/metals/pull/6226.

Also maybe it's worth to mention we're dropping Java 8 for running Metals, plugin authors might want to add proper error messages.